src/format/scan.rs

```diff
@@ -219,9 +219,48 @@ pub(super) fn space1(s: &str) -> &str {
     }
 }
 
-/// Consumes any number (including zero) of colon or spaces.
+/// Allow a colon with possible one-character whitespace padding.
+/// Consumes zero or one of these leading patterns:
+/// `":"`, `" "`, `" :"`, `": "`, or `" : "`.
 pub(super) fn colon_or_space(s: &str) -> ParseResult<&str> {
-    Ok(s.trim_left_matches(|c: char| c == ':' || c.is_whitespace()))
+    let c0s = match s.chars().next() {
```

Comment:
Potentially the following could be done a bit more simply? I've included one possibility below. Also I've included some commented error paths where we could potentially be a bit more specific, eg allowing only the specified patterns. If we prefer to keep it always returning ok, the function could potentially be renamed to something like `maybe_colon_or_space`

```rust
if s.is_empty() {
        // return Err(TOO_SHORT)
        return Ok(s)
    }

    fn next_char_is_whitespace(input: &str) -> bool {
        input.chars().next().map(|c| c.is_whitespace()).unwrap_or_default()
    }

    if s.starts_with(':') {
        s = s_next(s);
        if next_char_is_whitespace(s) {
            s = s_next(s);
        }
        return Ok(s)
    }
    
    
    if next_char_is_whitespace(s) {
        s = s_next(s);
        if s.starts_with(':') {
            s = s_next(s);
        } else {
            return Ok(s)
        }
        if next_char_is_whitespace(s) {
            s = s_next(s);
        }
        return Ok(s)
    }

    // Err(INVALID) 
    Ok(s)
```

Reply:
🤔

...

Yes, this fits with this PR, and would eliminate a few extra code paths. I will include this in the next update.

(I'm somewhat contradicting my prior comment https://github.com/chronotope/chrono/pull/807#discussion_r1021114078)

Reply:
This is in the latest PR push. https://github.com/chronotope/chrono/pull/807/commits/1e9f90bdf827699cef7dd6da41c37f3b95f9cdcd